docs/bioformats-release-process.rst

```diff
@@ -0,0 +1,257 @@
+BioFormats release
+==================
+
+.. _Bio-Formats: https://github.com/ome/bioformats
+.. _Bio-Formats Documentation: https://github.com/ome/bio-formats-documentation
+.. _Bio-Formats Downloads: https://downloads.openmicroscopy.org/bio-formats/
+.. _Octave Docker: https://github.com/ome/bio-formats-octave-docker
+.. _Read The Docs: https://readthedocs.org/projects/bio-formats/builds/
+.. _Data Repository: https://github.com/openmicroscopy/data_repo_config
+.. _Bio-Formats homebrew: https://github.com/ome/homebrew-alt
+
+This document describes the release process of the `Bio-Formats`_ Java library.
+The release process uses GitHub actions, make sure that the actions are active before pushing any tag.
+
+Before starting the release process, 
+open a Pull Request with a new release entry in the `whats-new <https://github.com/ome/bio-formats-documentation/blob/master/sphinx/about/whats-new.rst>`_ page. When approved by both the OME team and the Glencoe Software team, merge and start the release process.
+
+
+Release process
+^^^^^^^^^^^^^^^
+
+Source code release
+-------------------
+
+Before making a new release, merge all contributions on the ``develop`` branch of `Bio-Formats`_.
+
+The first operation to perform a Maven release is to bump the version out of
+``SNAPSHOT`` using the Maven versions plugin, add and commit::
+
+    $ mvn versions:set -DnewVersion=x.y.z -DgenerateBackupPoms=false
+    $ mvn versions:set-property -Dproperty=release.version -DnewVersion=x.y.z -DgenerateBackupPoms=false
+    $ git add -u .
+    $ git commit -m "Bump release version to x.y.z"
+
+
+A signed tag must be created for the released version using :command:`git tag -s`::
+
+    $ git tag -s -m "Tag version x.y.z" vx.y.z
+
+
+Revert to ``SNAPSHOT``, add and commit::
+
+    $ mvn versions:set -DnewVersion=x.y.t-SNAPSHOT -DgenerateBackupPoms=false
+    $ mvn versions:set-property -Dproperty=release.version -DnewVersion=x.y.z-SNAPSHOT -DgenerateBackupPoms=false
+    $ git add -u .
+    $ git commit -m "Revert to snapshot"
+
+Both the ``develop`` branch as well as the tag must be pushed upstream::
+
+    $ git push origin develop vx.y.z
+
+This will trigger GitHub action builds, generate artifacts, create a GitHub release and upload
+the artifacts to the release on GitHub and to the `OME artifactory`_. 
+An hourly cron job runs on our virtual machine and copy the artifacts published on Github under `Bio-Formats Downloads`_.
+
+Close the milestone if any and add new one if needed.
+
+
+Testing the artifacts
+---------------------
+
+To test the artifacts published, 
+ - Test the `bio-formats-command-line-tools <https://docs.openmicroscopy.org/internal/testing_scenarios/GeneralRelease.html#bio-formats-command-line-tools>`_.
```

Comment:
As these hyperlinks require (UoD?) authentication, I suspect a follow-up action might be to extract these testing scenarios to a public page?

Reply:
it is.
This is something that we need to extract. we faced the same issue during the Django upgrade testing